uploaded/file.go

```diff
@@ -4,14 +4,19 @@ import (
 	"mime/multipart"
 )
 
-// File represents a file and it's name & content-type
-type File struct {
+// FileMetadata represents the metadata that comes with an uploaded file
+type FileMetadata struct {
 	Filename    string
 	ContentType string
-	Data        multipart.File
 }
 
-// NewFile creates a new uploade file instance
-func NewFile(filename string, contentType string, file multipart.File) *File {
-	return &File{Filename: filename, ContentType: contentType, Data: file}
+// File represents a file and it's name & content-type
+type File struct {
+	Metadata FileMetadata
+	Data     multipart.File
+}
+
+// NewFile creates a new uploaded file instance
+func NewFile(metadata FileMetadata, data multipart.File) *File {
```

Comment:
can this go into `datautils/resource.go`, or worth keeping separate / here for some reason?

Reply:
how about `datautils/file.go` it's not related to the Resource code at all, this is something that I'd like to put in `models`, but since that's a generated class I can't add it there.